src/braket/devices/device.py

```diff
@@ -116,6 +116,18 @@ def status(self) -> str:
         """
         return self._status
 
+    def set_noise_model(self, noise_model: NoiseModel) -> None:
+        """Set the noise model of the device.
+
+        Args:
+            noise_model (NoiseModel): The Braket noise model to apply to the circuit before
+                execution. Noise model can only be added to the devices that support noise
+                simulation.
+        """
+        if noise_model:
+            self._validate_device_noise_model_support(noise_model)
+        self._noise_model = noise_model
```

Comment:
No need to do the `if` check if the method explicitly takes a noise model. As well, should this raise a warning if there's already a noise model attached? Alternatively, the method can be `update_noise_model`, in which case it will be pretty clear that the model will be overwritten.

Reply:
updated!

Reply:
Hi @speller26 could you review this PR again? I think it'd be good to get it merged before UnitaryHack